java/core/src/main/java/sleeper/core/statestore/FileReferenceStore.java

```diff
@@ -136,12 +136,25 @@ void atomicallyReplaceFileReferencesWithNewOne(String jobId, String partitionId,
      * Atomically updates the job field of file references, as long as the job field is currently unset. This will be
      * used for compaction job input files.
      *
-     * @param jobId          The job id which will be added to the {@link AllReferencesToAFile}
-     * @param fileReferences The {@link AllReferencesToAFile} whose status will be updated
+     * @param requests A list of {@link AssignJobIdRequest}s which should each be applied atomically
      * @throws FileReferenceNotFoundException      if a reference does not exist
      * @throws FileReferenceAssignedToJobException if a reference is already assigned to a job
      * @throws StateStoreException                 if the update fails for another reason
      */
+    default void atomicallyAssignJobIdsToFileReferences(List<AssignJobIdRequest> requests)
+            throws StateStoreException {
+        for (AssignJobIdRequest request : requests) {
+            atomicallyAssignJobIdToFileReferences(request.getJobId(),
+                    request.getFilenames().stream()
+                            .map(filename -> FileReference.builder()
+                                    .filename(filename)
+                                    .partitionId(request.getPartitionId())
+                                    .numberOfRecords(0L)
```

Comment:
Is it necessary to set the number of records here? We don't want to overwrite what's already in the store?

Reply:
See the changes to this file in the S3 state store implementation PR, where the default implementation is flipped since the new method signature is implemented everywhere:

https://github.com/gchq/sleeper/pull/1868/files#diff-789e19e19662ac88905db9ec9e80f1e4dbc96b29771f94049565692d9fee65ce